tests/rbd_mirror/test_rbd_mirror_daemon_restart_imagesync.py

```diff
@@ -0,0 +1,166 @@
+import time
+
+from ceph.parallel import parallel
+from tests.rbd_mirror.rbd_mirror_utils import rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+
+
+def run(**kw):
+    """
+    Verification of image sync continuation after multiple rbd mirror daemon restarts.
+    Args:
+        **kw: test data
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    Test case covered -
+    CEPH-10468 - Bug 1348940 - Restart of RBD daemon is again initiating full Sync/Copy of an Image
+    Pre-requisites -
+    Two ceph clusters with rbd mirror configured along with at least 1 client.
+
+    Test Case Flow:
+        1. Configure mirorring between two clusters.
+        2. induce decoy by adding images and io in background
+        3. Reduce concurrent image syncs to delay sync process to be able to verify with focus in
+           images under observation.
+        4. Initiate resync of images and restart rbd mirorring daemon in between.
+        5. Make sure that image sync doesn't start from the begining.
+    """
+    log.info("Starting RBD mirroring test case - CEPH-10468")
+
+    config = kw.get("config")
+
+    if not (config.get("rep_pool_config") or config.get("ec_pool_config")):
+        pool_image_config = {
+            "rep_pool_config": {
+                "mode": "pool",
+                "mirrormode": "journal",
+                "image": "rbd_image_rep",
+                "pool": "rbd_pool_10468",
+                "size": "25G",
+                "io_total": "10M",
+            },
+            "ec_pool_config": {
+                "pool": "rbd_image_pool_10468",
+                "image": "rbd_image_ec",
+                "mode": "image",
+                "data_pool": "data_pool",
+                "mirrormode": "snapshot",
+                "size": "25G",
+                "io_total": "10M",
+            },
+        }
+    kw["config"].update(pool_image_config)
+    # temp
+    mirror_obj = rbd_mirror_config(**kw)
+    # mirror1, mirror2 = [
+    #        rbdmirror.RbdMirror(cluster, config)
+    #        for cluster in kw.get("ceph_cluster_dict").values()
+    #    ]
+    # temp
+    mirror1 = mirror_obj["rep_rbdmirror"]["mirror1"]
+    mirror2 = mirror_obj["rep_rbdmirror"]["mirror2"]
+
+    rep_image_spec = (
+        f'{config["rep_pool_config"]["pool"]}/{config["rep_pool_config"]["image"]}'
+    )
+    ec_image_spec = (
+        f'{config["ec_pool_config"]["pool"]}/{config["ec_pool_config"]["image"]}'
+    )
+
+    def initiate_resync_and_wait_sync_to_start(img_spec):
```

Comment:
Reason for defining methods inside the run method?

Reply:
mainlu helps reducing number of **p.spawn** calls within parallel blocks which makes it more readable  and easily manageable by grouping logically similar lines.
I find it more readable.

Reply:
Okay. LGTM.